PWGCF/Flow/Tasks/FlowPbPbpikp.cxx

```diff
@@ -8,494 +8,297 @@
   {
-    if (ismanualDCAcut && !(candidate.isGlobalTrackWoDCA() && candidate.isPVContributor() && std::abs(candidate.dcaXY()) < cfgCutDCAxy && std::abs(candidate.dcaZ()) < cfgCutDCAz && candidate.itsNCls() > cfgITScluster && candidate.tpcNClsFound() > cfgTPCcluster)) {
-      return false;
+    std::array<int, 3> bayesprobs = {static_cast<int>(track.bayesPi()), static_cast<int>(track.bayesKa()), static_cast<int>(track.bayesPr())};
+    int bayesid = -1;
+    int prob = 0;
+
+    for (int i = 0; i < 3; ++i) {
+      if (bayesprobs[i] > prob && bayesprobs[i] > 80) {
+        bayesid = i;
+        prob = bayesprobs[i];
+      }
     }
-
-    return true;
+    return std::make_pair(bayesid, prob);
   }
 
-  template <typename T>
-  bool selectionPID(const T& candidate, int PID)
+  template <typename TTrack>
+  int GetBayesPIDIndex(TTrack track)
   {
-    if (candidate.pt() > 0.4) {
-      onlyTPC = false;
-    }
-
-    if (PID == 0) {
-      if (onlyTOF) {
-        if (candidate.hasTOF() && std::abs(candidate.tofNSigmaPi()) < nsigmaCutTOFPi) {
-          return true;
-        }
-      } else if (onlyTOFHIT) {
-        if (candidate.hasTOF() && std::abs(candidate.tofNSigmaPi()) < nsigmaCutTOFPi) {
-          return true;
-        }
-        if (!candidate.hasTOF() &&
-            std::abs(candidate.tpcNSigmaPi()) < nsigmaCutTPCPi) {
-          return true;
-        }
-      } else if (onlyTPC) {
-        if (std::abs(candidate.tpcNSigmaPi()) < nsigmaCutTPCPi) {
-          return true;
-        }
-      } else {
-        if (candidate.hasTOF() && (candidate.tofNSigmaPi() * candidate.tofNSigmaPi() + candidate.tpcNSigmaPi() * candidate.tpcNSigmaPi()) < (nsigmaCutCombined * nsigmaCutCombined)) {
-          return true;
-        }
-        if (!candidate.hasTOF() && std::abs(candidate.tpcNSigmaPi()) < nsigmaCutTPCPi) {
-          return true;
-        }
-      }
-    } else if (PID == 1) {
-      if (onlyTOF) {
-        if (candidate.hasTOF() && std::abs(candidate.tofNSigmaKa()) < nsigmaCutTOFKa) {
-          return true;
-        }
-      } else if (onlyTOFHIT) {
-        if (candidate.hasTOF() && std::abs(candidate.tofNSigmaKa()) < nsigmaCutTOFKa) {
-          return true;
-        }
-        if (!candidate.hasTOF() && std::abs(candidate.tpcNSigmaKa()) < nsigmaCutTPCKa) {
-          return true;
-        }
-      } else if (onlyTPC) {
-        if (std::abs(candidate.tpcNSigmaKa()) < nsigmaCutTPCKa) {
-          return true;
-        }
+    int maxProb[3] = {80, 80, 80};
+    int pidID = -1;
+    std::pair<int, int> idprob = GivebayesID(track);
+    if (idprob.first == 0 || idprob.first == 1 || idprob.first == 2) { // 0 = pion, 1 = kaon, 2 = proton
+      pidID = idprob.first;
+      float nsigmaTPC[3] = {track.tpcNSigmaPi(), track.tpcNSigmaKa(), track.tpcNSigmaPr()};
+      if (idprob.second > maxProb[pidID]) {
+        if (abs(nsigmaTPC[pidID]) > 3)
+          return 0;
+        return pidID + 1; // shift the pid by 1, 1 = pion, 2 = kaon, 3 = proton
```

Comment:
I would suggest using an `enum` for particle named constants and using it consistently
The function can return `-1` (`kWRONGID = -1`) in case the identification is not accepted